buildkite.go

```diff
@@ -142,31 +136,9 @@ func NewOpts(opts ...clientOpt) (*Client, error) {
 	return c, nil
 }
 
-// NewClient returns a new buildkite API client. As API calls require authentication
-// you MUST supply a client which provides the required API key.
-//
-// Deprecated: Use NewOpts instead.
-func NewClient(httpClient *http.Client) *Client {
-	baseURL, _ := url.Parse(DefaultBaseURL)
-
-	c := &Client{
-		client:    httpClient,
-		BaseURL:   baseURL,
-		UserAgent: DefaultUserAgent,
-	}
-
-	c.populateDefaultServices()
-
-	if c.client != nil {
-		if tokenAuth, ok := c.client.Transport.(*TokenAuthTransport); ok {
-			tokenAuth.APIHost = baseURL.Host
-		}
-
-		if basicAuth, ok := c.client.Transport.(*BasicAuthTransport); ok {
-			basicAuth.APIHost = baseURL.Host
-		}
-	}
-	return c
+// NewOpts is an alias for NewClient
+func NewOpts(opts ...clientOpt) (*Client, error) {
+	return NewClient(opts...)
 }
```

Comment:
Hear me out...

```go
func NewClientWithOpts(opts ...clientOpts) (*Client, error) {
    return NewOpts(opts)
}
```

A `NewClientWithOpts` which returns a NewOpts which in turn returns a NewClient.